### ISSUE TITLE
`dbg_macro` removes empty dbg! macro

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#9914]
### Summary

.

### Lint Name

dbg_macro

### Reproducer

I tried this code:

```rust
fn main() {
    let _: () = dbg!();
}
```

I saw this happen:
` cargo clippy --fix -- -Aclippy::all -Wclippy::dbg_macro`
```
    Checking clpy v0.1.0 (/tmp/clpy)
warning: failed to automatically apply fixes suggested by rustc to crate `clpy`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error: expected expression, found `;`
 --> src/main.rs:2:17
  |
2 |     let _: () = ;
  |                 ^ expected expression

error: aborting due to previous error

Original diagnostics will follow.

warning: `dbg!` macro is intended as a debugging tool
 --> src/main.rs:2:17
  |
2 |     let _: () = dbg!();
  |                 ^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#dbg_macro
  = note: requested on the command line with `-W clippy::dbg-macro`
help: ensure to avoid having uses of it in version control
  |
2 -     let _: () = dbg!();
2 +     let _: () = ;
  |

warning: `clpy` (bin "clpy") generated 1 warning (run `cargo fix --bin "clpy"` to apply 1 suggestion)
warning: failed to automatically apply fixes suggested by rustc to crate `clpy`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error: expected expression, found `;`
 --> src/main.rs:2:17
  |
2 |     let _: () = ;
  |                 ^ expected expression

error: aborting due to previous error

Original diagnostics will follow.

warning: `clpy` (bin "clpy" test) generated 1 warning (1 duplicate)
    Finished dev [unoptimized + debuginfo] target(s) in 0.48s
```

I expected to see this happen:

### Version

```text
rustc 1.67.0-nightly (c5d82ed7a 2022-11-19)
binary: rustc
commit-hash: c5d82ed7a4ad94a538bb87e5016e7d5ce0bd434b
commit-date: 2022-11-19
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```
